job.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// job package implements access to job manager grpc server.
+package job
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"google.golang.org/grpc"
+
+	pb "github.com/alexbrainman/job/internal/proto"
+)
+
+// OutputPipe implements io.Reader, so Job.StdoutPipe and
+// Job.StderrPipe can be used as io.Reader everywhere.
+type OutputPipe struct {
+	job        *Job
+	stream     pb.JobManager_OutputClient
+	outputtype pb.OutputReq_OutputType
+	buf        *bytes.Buffer
+}
+
+// newOutputPipe creates OutputPipe struct and initialise struct fields.
+func newOutputPipe(job *Job, outputtype pb.OutputReq_OutputType) *OutputPipe {
+	return &OutputPipe{
+		job:        job,
+		outputtype: outputtype,
+		buf:        new(bytes.Buffer),
+	}
```

Comment:
nit: Not using a pointer here makes it clear that it is owned by `OutputPipe`.
```suggestion
	buf        bytes.Buffer
}

// newOutputPipe creates OutputPipe struct and initialise struct fields.
func newOutputPipe(job *Job, outputtype pb.OutputReq_OutputType) *OutputPipe {
	return &OutputPipe{
		job:        job,
		outputtype: outputtype,
	}
```